spec/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume_spec.rb

```diff
@@ -0,0 +1,34 @@
+describe ManageIQ::Providers::IbmCloud::PowerVirtualServers::StorageManager::CloudVolume do
+
+  let(:ems_cloud) { FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_cloud) }
+  let(:ems_storage) { FactoryBot.create(:ems_ibm_cloud_power_virtual_servers_storage, :parent_ems_id => ems_cloud.id) }
+  let(:cloud_volume) { FactoryBot.create(:cloud_volume_ibm_cloud_power_virtual_servers, :ext_management_system => ems_storage ) }
+
+  describe "#validate_delete_volume" do
+  context "status is in-use" do
+    let(:status) { "in-use" }
+
+    it "validates unavailable" do
+      def validate_delete_volume
+        msg = validate_volume
+        return {:available => msg[:available], :message => msg[:message]} unless msg[:available]
+        if status == "in-use"
+          return validation_failed("Delete Volume", "Can't delete volume that is in use.")
+        end
+
+        {:available => true, :message => nil}
+      end
+    end
+  end
+
+  context "status is available"
+    let(:status) { "available" }
+
+    it "validates available" do
+      def validation_failed(operation, reason)
+        {:available => false, :message   => _("Validation failed for %{name} operation %{operation}. %{reason}") % {:name => name,:operation => operation, :reason => reason}}
+        expects(cloud_volume.validates_delete_volume['available']).to falsy
```

Comment:
This line looks awesome! But still needs some work:
- `expect()` is not plural
- [`validate_delete_volume`](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/blob/master/app/models/manageiq/providers/ibm_cloud/power_virtual_servers/storage_manager/cloud_volume.rb#L31) "validate" is not plural
- The return hash key is `:available`, so you need to access it as `validates_delete_volume[:available]`
- The "false" matches is `be_falsy` 
- I think we this test we actually want `be_truthy` - if the volume is available we want to allow deletion.